docs/design/api-plugin-model.md

```diff
@@ -0,0 +1,139 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/design/api-plugin-model.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+# Kubernetes API Plugin Model
+
+## Goals
+
+* Allow new APIs to be registered and serviced by third party components
+  dynamically at run time, not requiring a recompilation of any core kubernetes
+  component.
+* All state is stored together by apiserver; third party components can remain
```

Comment:
While it's true that this is a convenience, I'm not sure it is an essential goal. Our main goal is to facilitate extension of the core without recompiling and/or forking the project. If it were possible to launch etcd in a turnkey fashion or to use the same etcd (in the case that the extension were vetted by a cluster administrator), that would be a good start.